src/hashtables.jl

```diff
@@ -116,28 +129,69 @@ const indicesTable, sizeTable, posTable = generateTables()
 gc();
 
 
-## Utilities to get/set the maximum order and number of variables;
-## they reset the hash tables
-get_maxOrder() = _params_taylorN.maxOrder
-set_maxOrder(n::Int) = set_params_TaylorN(n, _params_taylorN.numVars)
-get_numVars() = _params_taylorN.numVars
-set_numVars(n::Int) = set_params_TaylorN(_params_taylorN.maxOrder, n)
-
-function set_params_TaylorN(order::Int, numVars::Int)
-    (order > 0 && numVars>=1) ||
-        error("Incorrect order (<0) or number of variables (>=1)")
-    order == _params_taylorN.maxOrder && numVars == _params_taylorN.numVars &&
-        return order, numVars
-    oldOrder = _params_taylorN.maxOrder
-    oldVars = _params_taylorN.numVars
-    global _params_taylorN = ParamsTaylorN(order, numVars)
-
-    resize!(indicesTable,order+1)
-    resize!(sizeTable,order+1)
-    resize!(posTable,order+1)
-    indicesTable[:], sizeTable[:], posTable[:] = generateTables()
-    gc();
-    # show_params_TaylorN()
-
-    return order, numVars
+get_variable_names() = _params_taylorN.variable_names
+set_variable_names{T<:String}(names::Vector{T}) = _params_taylorN.variable_names = names
+
+@doc doc"""`set_variables` sets the names and number of the Taylor variables,
+as well as the order of the Taylor expansion.""" ->
+
+function set_variables{T}(names::Vector{T}; order=6)
+
+    order >= 1 || error("Order must be at least 1")
+
+    num_vars = length(names)
+    num_vars >= 1 || error("Number of variables must be at least 1")
+
+    _params_taylorN.variable_names = names
+
+    if !(order == _params_taylorN.maxOrder && num_vars == _params_taylorN.numVars)
+        # if these are unchanged, no need to regenerate tables
+
+        _params_taylorN.maxOrder = order
+        _params_taylorN.numVars = num_vars
+
+        resize!(indicesTable,order+1)
+        resize!(sizeTable,order+1)
+        resize!(posTable,order+1)
+
+        indicesTable[:], sizeTable[:], posTable[:] = generateTables()
+        gc();
+    end
+
+    # return a list of the new variables
+    [taylorN_variable(i) for i in 1:get_numVars()]
 end
+
+
+function set_variables{T<:String}(names::T; order=6, numvars=-1)
+    variable_names = split(names)
+
+    if length(variable_names) == 1 && numvars >= 1
+        name = variable_names[1]
+        variable_names = [string(name, subscriptify(i)) for i in 1:numvars]
+    end
+
+    set_variables(variable_names, order=order)
+
+end
+
+
+
+
+
+# function set_params_TaylorN(order::Int, numVars::Int)
```

Comment:
We should erase all this unneeded things, instead of leaving them there commented.